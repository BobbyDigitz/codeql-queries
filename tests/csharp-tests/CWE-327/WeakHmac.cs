using System;
using System.IO;
using System.Text;
using System.Runtime.Serialization;
using System.Security.Cryptography;
using System.Security.Permissions;

public class StaticSalt
{
    public void Test1()
    {
        var key = Encoding.UTF8.GetBytes("TestPassword");

        // BAD: MD5
        var md5 = new HMACMD5();
        var md5_1 = new HMACMD5(key);

        // BAD: SHA1 (not really but worth reporting)
        var sha1 = new HMACSHA1();
        var sha1_2 = new HMACSHA1(key);

        // GOOD: SHA256
        var sha2 = new HMACSHA256(key);
        // GOOD: SHA384
        var sha3 = new HMACSHA384(key);
        // GOOD: SHA512
        var sha5 = new HMACSHA512(key);
    }
}