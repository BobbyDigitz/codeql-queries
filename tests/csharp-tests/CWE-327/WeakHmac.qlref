CWE-327/WeakHMac.ql