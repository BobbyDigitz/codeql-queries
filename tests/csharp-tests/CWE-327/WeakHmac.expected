edges
nodes
subpaths
#select
| WeakHmac.cs:15:19:15:31 | object creation of type HMACMD5 | Algs |
| WeakHmac.cs:16:21:16:36 | object creation of type HMACMD5 | Algs |
| WeakHmac.cs:19:20:19:33 | object creation of type HMACSHA1 | Algs |
| WeakHmac.cs:20:22:20:38 | object creation of type HMACSHA1 | Algs |
